raiden/tests/integration/network/transport/test_matrix_transport_assumptions.py

```diff
@@ -84,6 +98,175 @@ def test_assumption_matrix_userid(local_matrix_servers):
+        timeout_ms=SHORT_TIMEOUT_MS, latency_ms=SHORT_TIMEOUT_MS, first_sync=True
+    )
+
+    msg = "All clients share a room, presence information must be available"
+    assert tracker1.address_presence[signer1.address] == UserPresence.ONLINE, msg
+    assert tracker1.address_presence[signer2.address] == UserPresence.ONLINE, msg
+    assert tracker1.address_presence[signer3.address] == UserPresence.OFFLINE, msg
+    assert tracker2.address_presence[signer1.address] == UserPresence.ONLINE, msg
+    assert tracker2.address_presence[signer2.address] == UserPresence.ONLINE, msg
+    assert tracker2.address_presence[signer3.address] == UserPresence.OFFLINE, msg
+
+    msg_no_sync = "Client3 never calls sync, all presences must be unknown."
+    assert len(tracker3.address_presence) == 0, msg_no_sync
+
+    # Wait for PRESENCE_TIMEOUT to happen
+    gevent.sleep(PRESENCE_TIMEOUT + CI_LATENCY)
+
+    # Do a new sync, this time client1 must change status to offline
+    client2.blocking_sync(
+        timeout_ms=PRESENCE_TIMEOUT, latency_ms=SHORT_TIMEOUT_MS, first_sync=True
+    )
+    assert tracker2.address_presence[signer1.address] == UserPresence.OFFLINE, msg
+    assert tracker2.address_presence[signer2.address] == UserPresence.ONLINE, msg
+    assert tracker2.address_presence[signer3.address] == UserPresence.OFFLINE, msg
+
+
+def test_assumption_user_is_online_while_sync_is_blocking(local_matrix_servers):
+    """A user presence does not change while a /sync is blocking.
+
+    This assumption test makes sure a user is considered online while there is
+    one outstand /sync request. This is important because it guides the choice
+    of the `DEFAULT_TRANSPORT_MATRIX_SYNC_TIMEOUT` setting. Because the node is
+    considered online while the /sync is withstanding, then a large value is
+    better to avoid unecessary load to the Matrix server.
+    """
+    # This is the interval in seconds which a client must perform /sync calls
+    # to stay online.
+    PRESENCE_TIMEOUT = 30
+
+    # This timeout is used to *avoid* blocking the sync thread, otherwise we
+    # would have to generate events for the long-polling to return.
+    SHORT_TIMEOUT_MS = 1
+    LONG_TIMEOUT_MS = 60_000
+
+    tracker1 = PresenceTracker()
+    client1, signer1 = create_logged_in_client(local_matrix_servers[0])
+    client1.add_presence_listener(tracker1.presence_listener)
+
+    tracker2 = PresenceTracker()
+    client2, _ = create_logged_in_client(local_matrix_servers[0])
+    client2.add_presence_listener(tracker2.presence_listener)
+
+    room: Room = client1.create_room("test", is_public=True)
+    client2.join_room(room.aliases[0])
+
+    client2.blocking_sync(
+        timeout_ms=SHORT_TIMEOUT_MS, latency_ms=SHORT_TIMEOUT_MS, first_sync=True
+    )
+    client1.blocking_sync(
+        timeout_ms=SHORT_TIMEOUT_MS, latency_ms=SHORT_TIMEOUT_MS, first_sync=True
+    )
+
+    def long_polling():
+        timeout_secs = (LONG_TIMEOUT_MS - 100) / 1_000
```

Comment:
what does the `- 100` achieve?

Reply:
IIRC it fixes a race with scheduling, if the exact value is used for both the timeouts the test is flaky.